core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java

```diff
@@ -641,4 +646,27 @@ public static SocketAddress getDataPortSocketAddress(WorkerNetAddress netAddress
     return address;
   }
 
+  /**
+   * Test if the input address is serving an Alluxio service. This method make use of the
+   * Thrift protocol for performing service communication.
+   *
+   * @param address the network address to ping
+   * @param serviceName the Alluxio service name
+   * @throws UnauthenticatedException If the user is not authenticated
+   * @throws TTransportException If there is a protocol transport error
+   */
+  public static void pingService(InetSocketAddress address, String serviceName)
+          throws UnauthenticatedException, TTransportException {
```

Comment:
- indent is off
- can we not throw `TTransportException`? Alluxio code base is likely to move away from thrift. I don't want to leave thrift specific types in util method signatures.

Reply:
I see, how about throwing an IOException?